src/renderer/prefComponents/editor/config.js

```diff
@@ -4,24 +4,24 @@ export const endOfLineOptions = [{
   label: 'Default',
   value: 'default'
 }, {
-  label: 'Carriage return and Line feed(CRLF)',
+  label: 'Carriage return and line feed (CRLF)',
   value: 'crlf'
 }, {
-  label: 'Line feed(Lf)',
+  label: 'Line feed (LF)',
   value: 'lf'
 }]
 
 export const trimTrailingNewlineOptions = [{
```

Comment:
Referring to the 4 changes below this, I don’t know if in the app it is explicitly stated that these are newline settings outside the (old) strings, @Jocs are you able to provide a screenshot?

Reply:
I'm not understanding your comment, but I welcome the developers to explain the intention, in case I misunderstood.

Reply:
I was referring to the changes for trimming new lines, does it say outside the old strings that these settings refer to new lines?

Reply:
I think that the term *newline* is used in the original text to refer to existing line breaks that appear leading or trailing the text at the time that the file is opened. Strictly these elements are not *new lines*, as in being lines that have been recently created. Does this clarification address your question?

Reply:
Not really. I meant that without explaination that these settings refer to “new”lines, 'Trim all leading and trailing' wouldn’t really make any sense, and I don’t know if the UI specifies that these settings refer to “new”lines outside the strings you changed.

Reply:
The significance of a list of options only needs to be clear in the context of the label that indicates what question is being answered by those options. To evaluate clarity and accuracy both the menu label and the menu options must be seen together.

Reply:
The developers will have to clarify. If the string is obsolete then maybe a cleanup commit could be provided upstream.

Reply:
The first two seem fine to keep. I preferred "Do nothing" over "Disabled" because the question is "What to do". It's not fully clear to me what the third one is intended to mean in terms of actual application behavior.

Reply:
> I preferred "Do nothing" over "Disabled" because the question is "What to do".

:+1:

> It's not fully clear to me what the third one is intended to mean in terms of actual application behavior.

We scan the document ending and apply one of the other options. If the document ends with a single newline, use `Ensure single trailing newline`, and if we detect no newline we use `Trim trailing newlines`. Otherwise we do nothing and keep the document ending as it is.

E.g.

```
line 1
line 2
...
line n
<empty line, just one newline>
```

--> Mark Text will ensure a single trailing newline before saving the document.

Reply:
> If the document ends with a single newline, use `Ensure single trailing newline`,



> and if we detect no newline we use `Trim trailing newlines`.


I think there is piece missing. Both statements seem to be null operations, because the test is the same as the effect.


Maybe it means remove any  extra newline characters in excess of one?

Reply:
You right, we restore the old text ending when the user edits the document and inserts more newlines.

1. Load file, check text ending and save option.
2. Before saving, we normalize the text ending depending on the option.
3. Save file with the same ending as we loaded it.